substrate/utility

```diff
@@ -8,7 +8,6 @@
 #include <array>
 #include <memory>
 #include <string>
-#include <limits>
```

Comment:
Does this mean to remove this header include?

Reply:
clangd warns that the header is unused. Should I revert?

Reply:
Ah, `std::numeric_limits` is the only thing that we ever use that comes from that header and it is indeed not used in this file, so that's fine - was just unsure if it was accidental or not, it's not so 👍🏼